### PR TITLE
Fixes related to using public APIs introduced in GraalVM 22.3

### PR DESCRIPTION
**Pull request by zakkak**
* Fixes a class cast exception by actually loading the class and not just its name.
* Reduces the invocations to Version.getCurrent() in the generated native-image Feature
* Specifies (as required) the module name when registering jaxp resource bundles

### CODE REVIEW

**Comment by zakkak** [comment]
@jerboaa can you please review?

File: core/deployment/src/main/java/io/quarkus/deployment/steps/NativeImageFeatureStep.java [hunk 1]
```diff
@@ -185,7 +185,7 @@ public void write(String s, byte[] bytes) {
                     TryBlock tryBlock = greaterThan22_2.tryBlock();
 
                     tryBlock.invokeStaticMethod(registerLambdaCapturingClass,
-                            tryBlock.load(i.getClassName()));
+                            tryBlock.loadClassFromTCCL(i.getClassName()));
```
Comment (jerboaa) [inline core/deployment/src/main/java/io/quarkus/deployment/steps/NativeImageFeatureStep.java:188]: Right, this is fixing: https://github.com/graalvm/mandrel/actions/runs/3156296401/jobs/5136091100#step:11:316

OK.

File: extensions/jaxp/deployment/src/main/java/io/quarkus/jaxp/deployment/JaxpProcessor.java [hunk 2]
```diff
@@ -30,15 +31,16 @@ void reflectiveClasses(BuildProducer<ReflectiveClassBuildItem> reflectiveClass)
 
     @BuildStep
     void resourceBundles(BuildProducer<NativeImageResourceBundleBuildItem> resourceBundle) {
+        Consumer<String> resourceBundleItemProducer = bundleName -> resourceBundle
+                .produce(new NativeImageResourceBundleBuildItem(bundleName, "java.xml"));
         Stream.of(
```
Comment (jerboaa) [inline extensions/jaxp/deployment/src/main/java/io/quarkus/jaxp/deployment/JaxpProcessor.java:36]: These changes are fixing: https://github.com/graalvm/mandrel/actions/runs/3156296401/jobs/5136091100#step:11:343, but we also need https://github.com/oracle/graal/pull/5118 correct?

**Review by jerboaa — APPROVED** [review]
Seems fine to me.

File: extensions/jaxp/deployment/src/main/java/io/quarkus/jaxp/deployment/JaxpProcessor.java (hunk 2 above)
Comment (zakkak) [reply extensions/jaxp/deployment/src/main/java/io/quarkus/jaxp/deployment/JaxpProcessor.java:36]: Correct

**Review by gsmet — APPROVED** [review]

**Pull request merged**